interpreter/assign/assign.go

```diff
@@ -134,9 +134,6 @@ func Assign(left, right value.Value) error {
 			rv := value.Unwrap[*value.Time](right)
 			lv.Value = rv.Value.Format(http.TimeFormat)
 		case value.BackendType: // STRING = BACKEND
-			if right.IsLiteral() {
-				return errors.WithStack(fmt.Errorf("BACKEND identifier could not assign to STRING"))
-			}
```

Comment:
The literal check of the right expression is still needed. `IsLiteral()` method returns bool that the right expression is a literal or not.

This case checks BACKEND literal assignment for STRING variable the following example:

```vcl
set req.backend = F_origin_0;
declare local var.backend STRING;
[VALID] set var.backend = req.backend; // req.backed is not a literal, should do type conversion
[INVALID] set var.backend = F_origin_0; // F_origin_0 is a literal
```

The above code is invalid in Fastly. I put a fiddle here: https://fiddle.fastly.dev/fiddle/108043ab